common/utils/src/main/resources/error/error-classes.json

```diff
@@ -840,6 +840,59 @@
       "Detected an incompatible DataSourceRegister. Please remove the incompatible library from classpath or upgrade it. Error: <message>"
     ]
   },
+  "INCOMPATIBLE_DATA_FOR_TABLE" : {
+    "message" : [
+      "Cannot write incompatible data for table <tableName>:"
+    ],
+    "subClass" : {
+      "AMBIGUOUS_COLUMN_NAME" : {
+        "message" : [
+          "Ambiguous column name in the input data: <colPath>."
```

Comment:
```suggestion
          "Ambiguous column name in the input data <colPath>."
```

Reply:
BTW, why do you use the term `path` instead of `name`. Let's be consistent and use `colName` (see other error classes).

Reply:
I used the name `comPath` as they are used in the function where the error actually occurs. e.g.

```scala
  private def resolveColumnsByPosition(
      tableName: String,
      inputCols: Seq[NamedExpression],
      expectedCols: Seq[Attribute],
      conf: SQLConf,
      addError: String => Unit,
      colPath: Seq[String] = Nil): Seq[NamedExpression] = {

      ...

      throw QueryCompilationErrors.incompatibleDataToTableExtraStructFieldsError(
        tableName, colPath.quoted, extraColsStr
      )
```


However, it seems that `colName` would be a better choice. I will make the necessary modifications.